### ISSUE TITLE
Coverlet doesn't track coverage of method that throws exception

### ISSUE THREAD

**Issue by diegofox2** [coverlet-coverage/coverlet#890]
Hi guys, thanks for maintaining this wonderful tool to measure code coverage.

In this image, you can see the coverage result for an IF statement that throws an exception inside when a condition happens.

![Partial Coverage](https://user-images.githubusercontent.com/7718960/86540046-23240d00-bed8-11ea-844e-f95a9bb9d82d.JPG)

And I have these test methods that fully cover the method AddSkillRequired.

![image](https://user-images.githubusercontent.com/7718960/86540095-b6f5d900-bed8-11ea-92a9-7cd3c96a2382.png)

As you can see, the code coverage report indicates that the IF statement is partially covered but it really has full coverage.

Am I doing something wrong or is there an issue here?

**Comment by MarcoRossignoli** [comment]
Thanks for reporting, could be a bug, need some investigation. Are you able to generate a repro?

**Comment by diegofox2** [comment]
Hi @MarcoRossignoli 

I'm sorry but I don't know what do you mean with "generate a repro". Can you elaborate, please?

**Comment by MarcoRossignoli** [comment]
I mean create a simple app that reproduce the issue so we can investigate with a real sample, you can create a sample repo and link here like this sample of other guy https://github.com/chtoucas/DeterministicBuilds

**Comment by MarcoRossignoli** [comment]
@diegofox2 I did some test but cannot repro in my local, this is my test https://github.com/coverlet-coverage/coverlet/compare/master...MarcoRossignoli:issue_890 and result
![image](https://user-images.githubusercontent.com/7894084/87228427-7519d680-c3a1-11ea-9315-6392319cbaf9.png)

**Comment by diegofox2** [comment]
Ok, I will create it tomorrow. Thank you!

**Comment by diegofox2** [comment]
Ok, as far I can see the problem is that I have two test projects generating cobertura files, and in the CoverletOutput directory the second file replaces the first one. That is why I never see the coverage of the first one.
I was reading that exists the parameter /`p:MergeWith` but is is not valid for cobertura files. Am I right?

**Comment by tonerdo** [comment]
@diegofox2 yeah. `/p:MergwWith` can only be used with Coverlet's own native format

**Comment by diegofox2** [comment]
Ok, thank you guys and sorry for consuming your time. One last question: is this possible to contribute to adding a feature that lets convert converlet native format into cobertura format? @tonerdo @MarcoRossignoli

**Comment by diegofox2** [comment]
I'm sorry, I downloaded your source code and after analyzing deeper the documentation, I realized that what I want to do is already available.
In this [document](https://github.com/coverlet-coverage/coverlet/blob/master/Documentation/Examples/MSBuild/MergeWith/HowTo.md), you explain how to use parameter `p:MergeWith` .

This is the original example:

`dotnet test /p:CollectCoverage=true  /p:CoverletOutput=../CoverageResults/ /p:MergeWith="../CoverageResults/coverage.json" /p:CoverletOutputFormat=\"opencover,json\" -m:1`

If you replace the value `opencover` for `cobertura` you will have the json file created but also a cobertura file that was built with the same information than the json file. 

`dotnet test /p:CollectCoverage=true  /p:CoverletOutput=../CoverageResults/ /p:MergeWith="../CoverageResults/coverage.json" /p:CoverletOutputFormat=\"cobertura,json\" -m:1`

So it is perfectly possible to merge the coverage result in a json file and then "export" it in another supported format like cobertura.

Thank you very much!
